File: cpp/program10.cpp
// program135.c

#include<iostream>
using namespace std;

class Pattern
{
    public:
       int iRow;
       int iCol;

    Pattern(int iNo1, int iNo2)
    {
        iRow = iNo1;
        iCol = iNo2;
    }

    void DisplayPattern()
    {
        int i = 0;
        int j = 0;

        if(iRow != iCol)
        {
            cout<<"Invalid Input"<<"\n";
            return;
        }

        for(i = 1; i <= iRow; i++)
        {
            for(j = 1; j <= iCol; j++ )
            {
                if(i == j)
                {
                    cout<<"**"<<"\t";
                }
                else if(i >= j)
                {
                    cout<<"{}"<<"\t";
                }
                else
                {
                    cout<<"()"<<"\t";
                }
            }
            cout<<"\n";
        }
    }
};

int main()
{
    int iValue1 = 0;
    int iValue2 = 0;

    cout<<"Enter number of Rows"<<"\n";
    cin>>iValue1;

    cout<<"Enter number of colums"<<"\n";
    cin>>iValue2;

    Pattern pobj(iValue1,iValue2);
    pobj.DisplayPattern();

    return 0;
}

File: java/program10.java
import java.util.*;

class Pattern
{
    public int iRow;
    public int iCol;

    public Pattern(int X, int Y)
    {
        iRow = X;
        iCol = Y;
    }

    public void Display()
    {
        int i = 0, j = 0;

        if(iRow != iCol)
        {
            cout<<"Invalid Input"<<"\n";
            return;
        }

        for(i = 1; i <= iRow; i++)
        {
            for(j = 1; j <= iCol; j++)
            {
                if(i == j)
                {
                    System.out.print("(^_^)\t");
                }
                else if(i <= j)
                {
                    System.out.print("(*_*)\t");
                }
                else
                {
                    System.out.print("(~_~)\t");
                }
            }
            System.out.println();
        }
    }
}

class program10
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number of row");
        int iValue1 = sobj.nextInt();

        System.out.println("Enter number of column");
        int iValue2 = sobj.nextInt();

        Pattern obj = new Pattern(iValue1,iValue2);
        obj.Display();
    }
}